### ISSUE TITLE
Label 'for' attribute doesn't match referenced element ID on generated entities

### ISSUE THREAD

**Issue by mandudeboy** [jhipster/generator-jhipster#9561]
##### **Overview of the issue**

Generated "create new entity" pages in React have labels for each input field in the form. The labels are not clickable which is an accessibility issue. This is due to labels "for" field not being used correctly.

##### **Motivation for or Use Case**

Using assistive technologies means you rely on correctly implemented HTML. Labels that are not connected properly to the field they reference mean screen readers won't read the content correctly, and users cannot click the label to focus the input field (especially common use for checkbox fields)

##### **Reproduce the error**

Generate any entity in React. Go to entity page, press "create new entity". Try clicking a label to focus an input field. (The Boolean type input labels may 'feel' clickable, but that is only because the checkbox has been made so large that it covers the label element, it is still incorrectly read by screen readers.)

##### **Related issues**

Only labels for registration views and other static views have been reported and fixed, but not generated entities.

##### **Suggest a Fix**

This is example of generated code:
`<AvGroup>
                    <Label for="id">
                      <Translate contentKey="global.field.id">ID</Translate>
                    </Label>
                    <AvInput id="card-id" type="text" className="form-control" name="id" required readOnly />
                  </AvGroup>`

should be replaced by:
`<AvGroup>
                    <Label for="card-id">
                      <Translate contentKey="global.field.id">ID</Translate>
                    </Label>
                    <AvInput id="card-id" type="text" className="form-control" name="id" required readOnly />
                  </AvGroup>`

##### **JHipster Version(s)**

Generated using v5.4.2

##### **JHipster Version(s)**

```
jobbsam@0.0.0 /home/joel/Documents/git/respekttrappan
└── generator-jhipster@5.8.1 

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "se.axesslab.respekttrappan",
      "nativeLanguage": "sv"
    },
    "jhipsterVersion": "5.4.2",
    "applicationType": "monolith",
    "baseName": "respekttrappan",
    "packageName": "se.axesslab.respekttrappan",
    "packageFolder": "se/axesslab/jobbsam",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "secretkey",
    "clientFramework": "react",
    "useSass": true,
    "clientPackageManager": "npm",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "sv",
    "languages": [
      "sv",
      "da",
      "en"
    ],
    "herokuAppName": "*****-test",
    "herokuDeployType": "git"
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Board {
  type BoardType required,
  name String required,
  isTemplate Boolean
}
entity Lane {
  name String required,
  pos Integer required
}
entity Card {
  name String required,
  description String,
  type CardType,
  points Integer,
  pos Integer required,
  dueDate Instant,
  dueCompleted Boolean,
  assignee String,
  quizUrl String maxlength(10000),
  isForGroups Boolean,
  timeRequired String
}
entity Label {
  name String required
}
entity Comment {
  text String required,
  createdDate Instant required,
  updatedDate Instant
}
entity Attachment {
  name String,
  type AttachmentType,
  url String required,
  thumbnailUrl String,
  createdDate Instant required,
  pos Integer required
}
entity Task {
  name String required,
  done Boolean
}
entity Document {
  name String required,
  description String,
  url String required,
  lastModified Instant required
}

enum BoardType {
  DYSLEXIA
}

enum CardType {
  AWARENESS,
  ANALYSIS,
  IMPLEMENTATION,
  STABILITY
}

enum AttachmentType {
  LINK,
  DOCUMENT,
  IMAGE
}

relationship OneToMany {
  Board{lane} to Lane{board(name)},
  Lane{card} to Card{lane(name)},
  Card{clone} to Card{template(name)},
  Card{comment} to Comment{card(name)},
  Card{attachment} to Attachment{card(name)},
  Card{task} to Task{card(name)}
}
relationship ManyToOne {
  Board{owner(login)} to User,
  Comment{author(login)} to User,
  Document{owner(login)} to User
}
relationship ManyToMany {
  Card{label(name)} to Label{card},
  Card{document(name)} to Document{card}
}

dto Board, Lane, Card, Label, Comment, Attachment, Task, Document with mapstruct
paginate Board, Lane, Card, Label, Comment, Attachment, Task, Document with pagination
service Board, Lane, Card, Label, Comment, Attachment, Task, Document with serviceClass

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

git version 2.7.4

node: v9.4.0

npm: 6.4.1

yeoman: 2.0.5

yarn: 1.12.3

Docker version 18.09.0, build 4d60db4

##### **Browsers and Operating System**

-   [ x] Checking this box is mandatory (this is just to show you read everything)

**Comment by wmarques** [comment]
@mandudeboy thanks for the report, indeed the `for` need to match the input `id`
Are you willing to do a PR to fix this ? Shouldn't be a big change
